### PR TITLE
[Storage] Restricted mode banner and flow

### LINKED ISSUE #2020: [Storage] Add support for restricted accounts
Implement restricted account UI elements (banner and disable uploads)

### PR DESCRIPTION
**Pull request by Tbaut**
closes #2020

---
Submission checklist: 

   

#### Layout
- [x] Change looks good in the desktop web ui

### CODE REVIEW

File: packages/storage-ui/src/Contexts/StorageApiContext.tsx [hunk 1]
```diff
@@ -228,6 +230,22 @@ const StorageApiProvider = ({ apiUrl, withLocalStorage = true, children }: Stora
...
+    if (accessToken && accessToken.token && storageApiClient) {
+      storageApiClient?.setToken(accessToken.token)
+      const decodedAccessToken = jwtDecode<{ perm: { secured?: string; files?: string } }>(
+        accessToken.token
+      )
+
+      if (decodedAccessToken.perm.files === "restricted") {
+        setAccountRestricted(true)
+      } else {
+        setAccountRestricted(false)
```
Comment (Tbaut) [inline packages/storage-ui/src/Contexts/StorageApiContext.tsx:244]: FYI @asnaith, change this to `true` to test, and your account will be in restricted mode.

File: packages/storage-ui/src/Components/Pages/BucketPage.tsx [hunk 2]
```diff
@@ -160,9 +160,19 @@ const BucketPage: React.FC<IFileBrowserModuleProps> = () => {
 
   const handleUploadOnDrop = useCallback(async (files: File[], fileItems: DataTransferItemList, path: string) => {
     if (!bucket) return
+
+    if (accountRestricted) {
+      addToast({
+        type:"error",
+        title: t`Uploads disabled`,
+        subtitle: t`Oops! You need to pay for this month to upload more content.`
```
Comment (tanmoyAtb) [inline packages/storage-ui/src/Components/Pages/BucketPage.tsx]: Something in the line of "Please clear due payments to upload more content" like what we have in the restricted banner sounds more appropriate to me.

**Review by tanmoyAtb — APPROVED** [review]
Looks great :)

File: packages/storage-ui/src/Components/Pages/BucketPage.tsx (hunk 2 above)
Comment (Tbaut) [reply packages/storage-ui/src/Components/Pages/BucketPage.tsx]: What do you think about
```suggestion
        subtitle: t`Your account is restricted. Until you&apos;ve settled up, you can&apos;t upload any new content`
```

Comment (tanmoyAtb) [reply packages/storage-ui/src/Components/Pages/BucketPage.tsx]: Yup. This is better.

Comment (FSM1) [reply packages/storage-ui/src/Components/Pages/BucketPage.tsx]: Can you apply the same update to Files while you're at it?

File: packages/storage-ui/src/Contexts/StorageApiContext.tsx (hunk 1 above)
Comment (asnaith) [reply packages/storage-ui/src/Contexts/StorageApiContext.tsx:244]: @Tbaut  awesome. Thank you for this tip 👍

**Comment by asnaith** [comment]
With the modification of the code, I was able to see the banner and the disabled bucket and pin buttons 👍

The hover color of the button is the same as the banner background. 
Screen Shot 2022-03-21 at 2 34 35 PM

As I am forcing restricted mode through code and not the api response I am unable to test the full functionality and see if the banner disappears and the buttons are re-enabled when a user re-adds a card/upgrades their subscription. 

I'm wondering if there is any additional info displayed to the user on the settings page, informing them of the specific action they need to take? I don't see additional elements on the settings page and expected something to further explain the reason for the restriction / required action but perhaps because I'm using the code mod to test and not genuine api responses for a restricted user etc.

File: packages/storage-ui/src/Components/Pages/BucketPage.tsx (hunk 2 above)
Comment (Tbaut) [reply packages/storage-ui/src/Components/Pages/BucketPage.tsx]: yup, that was my plan :D

**Comment by FSM1** [comment]
> With the modification of the code, I was able to see the banner and the disabled bucket and pin buttons 👍
> 
> The hover color of the button is the same as the banner background. Screen Shot 2022-03-21 at 2 34 35 PM
> 
> As I am forcing restricted mode through code and not the api response I am unable to test the full functionality and see if the banner disappears and the buttons are re-enabled when a user re-adds a card/upgrades their subscription.
> 
> I'm wondering if there is any additional info displayed to the user on the settings page, informing them of the specific action they need to take? I don't see additional elements on the settings page and expected something to further explain the reason for the restriction / required action but perhaps because I'm using the code mod to test and not genuine api responses for a restricted user etc.

There is currently no additional info displayed. The user would see an unpaid invoice on the `settings/billing` and I feel that this should be enough for now.

**Comment by Tbaut** [comment]
@asnaith re the hover background, the fact that the text remains white, and therefor visible makes me this that it isn't really a problem :shrug: Let me know your thoughts.

Agreed with Mike that the unpaid invoice (in red) that is visible when you've something to pay, should be enough.

**Review by FSM1 — APPROVED** [review]

**Comment by asnaith** [comment]
> @asnaith re the hover background, the fact that the text remains white, and therefor visible makes me this that it isn't really a problem 🤷 Let me know your thoughts.
> 

@Tbaut yeah it's extremely minor and agree it's not a problem at all as it stays visible and functions as intended. I was just pointing it out in case we wanted to add that extra layer of polish to the presentation (I assumed changing it would be as simple as just changing a color hex).

**Pull request merged**